src/control/build/interop.go

```diff
@@ -114,13 +130,15 @@ var defaultRules = []*InteropRule{
 	},
 }
 
-var (
-	Server22xAgent20x = &InteropRule{
+// releaseRules are a set of rules which apply to a specific release,
+// and are likely to change between releases.
+var releaseRules = []*InteropRule{
+	{
 		Self:          ComponentServer,
 		Other:         ComponentAgent,
 		Description:   "server v2.2.x is compatible with agent v2.0.x",
 		StopOnSuccess: true,
-		Check: func(self, other *VersionedComponent) bool {
+		Match: func(self, other *VersionedComponent) bool {
```

Comment:
A little confused about the motivation for adding Match vs. Check. Logically are these things really different?

Reply:
This is a refinement of the work landed previously on this branch and master. What I found on the master branch was that the previous approach to determining if a rule should be processed was too simple, so I added this additional method to provide for custom matching logic if the simple logic isn't adequate. Now there is a default `Match()` method that implements the simple logic, and it's called first to determine if the rule is relevant before either skipping or proceeding to call `Check()`.

Reply:
Thanks for the explanation. Everything else looks good.